source/MaterialXRender/Handlers/ImageHandler.cpp

```diff
@@ -64,18 +66,23 @@ bool ImageHandler::saveImage(const FilePath& filePath,
                              bool verticalFlip)
 {
     FilePath foundFilePath = findFile(filePath);
+    if (foundFilePath.isEmpty())
+    {
+        return false;
+    }
 
-    std::pair <ImageLoaderMap::iterator, ImageLoaderMap::iterator> range;
-    string extension = MaterialX::getFileExtension(foundFilePath);
-    range = _imageLoaders.equal_range(extension);
-    ImageLoaderMap::iterator first = --range.second;
-    ImageLoaderMap::iterator last = --range.first;
-    for (auto it = first; it != last; --it)
+    string extension = foundFilePath.getExtension();
```

Comment:
Previous code would crash on Mac for some reason. This new code passes CI and should be less error prone as guessing the iterator decrements might have been the cause of the issues.